modules/custom/social_graphql/src/GraphQL/EntityConnection.php

```diff
@@ -351,4 +394,30 @@ protected function assertValidPagination(?int $first, ?string $after, ?int $last
     }
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  public function getMetaData(): CacheableMetadata {
+    // We want to make sure that metadata is set before it is fetched.
+    // This makes sure that proper cachability actions can be performed.
+    if ($this->metadata === NULL) {
+      // @todo Create a custom exception.
```

Comment:
This is something we should still do in the PR. It's small enough that it doesn't need a follow-up.